packages/safe-ds-eda/src/webviewState.ts

```diff
@@ -26,6 +26,32 @@ window.addEventListener('message', (event) => {
             // This should be fired immediately whenever the panel is created or made visible again
             currentState.set(message.value);
             break;
+        case 'setProfiling':
+            if (get(currentState) && get(currentState).table) {
+                currentState.update((state) => {
+                    return {
+                        ...state,
+                        table: {
+                            ...state.table!,
+                            columns: state.table!.columns.map((column) => {
```

Comment:
A flatter state is easier to update and reason about. Here, you could store profiling information as a top-level item in the state as an object that maps column names to profiling information. Then you can quickly set the profiling information given a column name.

Reply:
but then when I display the profiling in my HTML I will have to iterate through that Profiling info and can not anymore just access it by the column that I am iterating over anyway. The Svelte stores that I use (particularly the history) will be split up a bit more in the future, so that I can update only a part and not all HTML gets rerendered on say an addition to the history. So this will be addressed in another capacity, but afraid that in my opinion this profiling still belongs here with the column object and the state is not planned to get flatter itself, only the handling of it.